samples/extensions/dynamic_rendering/dynamic_rendering.cpp

```diff
@@ -0,0 +1,466 @@
+			                                 0,
+			                                 VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT,
+			                                 VK_IMAGE_LAYOUT_UNDEFINED,
+			                                 VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL,
+			                                 VK_PIPELINE_STAGE_TOP_OF_PIPE_BIT,
+			                                 VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT, range);
+
+			vkb::insert_image_memory_barrier(draw_cmd_buffer,
+			                                 depth_stencil.image,
+			                                 0,
+			                                 VK_ACCESS_DEPTH_STENCIL_ATTACHMENT_WRITE_BIT,
+			                                 VK_IMAGE_LAYOUT_UNDEFINED,
+			                                 VK_IMAGE_LAYOUT_DEPTH_ATTACHMENT_OPTIMAL,
+			                                 VK_PIPELINE_STAGE_TOP_OF_PIPE_BIT,
+			                                 VK_PIPELINE_STAGE_VERTEX_SHADER_BIT, depth_range);
+
+			VkRenderingAttachmentInfoKHR color_attachment_info = vkb::initializers::rendering_attachment_info();
+			color_attachment_info.imageView                    = swapchain_buffers[i].view;        // color_attachment.image_view;
+			color_attachment_info.imageLayout                  = VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL;
+			color_attachment_info.resolveMode                  = VK_RESOLVE_MODE_NONE;
+			color_attachment_info.loadOp                       = VK_ATTACHMENT_LOAD_OP_CLEAR;
+			color_attachment_info.storeOp                      = VK_ATTACHMENT_STORE_OP_STORE;
+			color_attachment_info.clearValue                   = clear_values[0];
+
+			VkRenderingAttachmentInfoKHR depth_attachment_info = vkb::initializers::rendering_attachment_info();
+			depth_attachment_info.imageView                    = depth_stencil.view;
+			depth_attachment_info.imageLayout                  = VK_IMAGE_LAYOUT_DEPTH_ATTACHMENT_OPTIMAL;
+			depth_attachment_info.resolveMode                  = VK_RESOLVE_MODE_NONE;
+			depth_attachment_info.loadOp                       = VK_ATTACHMENT_LOAD_OP_CLEAR;
+			depth_attachment_info.storeOp                      = VK_ATTACHMENT_STORE_OP_STORE;
+			depth_attachment_info.clearValue                   = clear_values[1];
+
+			auto render_area               = VkRect2D{VkOffset2D{}, VkExtent2D{width, height}};
+			auto render_info               = vkb::initializers::rendering_info(render_area, 1, &color_attachment_info);
+			render_info.layerCount         = 1;
+			render_info.pDepthAttachment   = &depth_attachment_info;
+			render_info.pStencilAttachment = &depth_attachment_info;
+
+			vkCmdBeginRenderingKHR(draw_cmd_buffer, &render_info);
+			draw_scene();
+			vkCmdEndRenderingKHR(draw_cmd_buffer);
+
+			vkb::insert_image_memory_barrier(draw_cmd_buffer,
+			                                 swapchain_buffers[i].image,
+			                                 VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT,
+			                                 VK_ACCESS_COLOR_ATTACHMENT_READ_BIT,
```

Comment:
No dstAccessMask needed here; semaphore takes care of availability. (And BOTTOM + COLOR_ATTACHMENT_READ_BIT is either way invalid usage.)